near/omni-types/src/lib.rs

```diff
@@ -360,16 +376,35 @@ impl Fee {
     }
 }
 
-pub type TransferId = (ChainKind, Nonce);
+#[derive(
+    BorshDeserialize,
+    BorshSerialize,
+    Serialize,
+    Deserialize,
+    Debug,
+    Clone,
+    PartialEq,
+    Eq,
+    Default,
+    Copy,
+)]
+pub struct TransferId {
+    // The origin chain kind
+    pub chain: ChainKind,
+    // The transfer nonce that maintained on the source chain
+    pub nonce: Nonce,
```

Comment:
I prefer to make it explicit as well:

```suggestion
    pub origin_nonce: Nonce,
```

Reply:
a9cb73ce0b057e2e194c5865f7d262c3404bea47